keylime/tpm/tpm_main.py

```diff
@@ -1157,13 +1217,26 @@ def check_quote(
         if hash_alg is None:
             hash_alg = self.defaults["hash"]
 
+        # First and foremost, the quote needs to be validated
         retout, success = self._tpm2_checkquote(aikTpmFromRegistrar, quote, nonce, hash_alg, compressed)
         if not success:
             # If the quote validation fails we will skip all other steps therefore this failure is irrecoverable.
             failure.add_event(
-                "quote_validation", {"message": "Quote validation using tpm2-tools", "data": retout}, False
+                "quote_validation", {"message": "Quote data validation using tpm2-tools", "data": retout}, False
+            )
+            return failure
+
+        # Only after validating the quote, the TPM clock information can be extracted from it.
+        clock_failure, clock_info = self.check_quote_timing(clock_info, quote)
```

Comment:
Maybe name the parameter `previous_clock_info` to avoid confusion.

Reply:
+1